src/ontodia/diagram/halo/haloMarkup.tsx

```diff
@@ -0,0 +1,49 @@
+import * as joint from 'jointjs';
+import * as React from 'react';
+
+export interface Props {
+    cellView: joint.dia.CellView;
+    onDelete: () => void;
+    onExpand: () => void;
+}
+
+export interface State {
+    cellIsExpanded: boolean;
+}
+
+export class HaloMarkup extends React.Component<Props, State> {
+
+    constructor(props: Props) {
+        super(props);
+        this.state = {
+            cellIsExpanded: this.props.cellView.model.get('isExpanded'),
```

Comment:
Instead of duplicating `isExpanded` state in the component (which is error-prone), you may just pass it as a prop.